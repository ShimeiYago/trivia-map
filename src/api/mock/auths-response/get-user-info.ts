import { User } from 'types/user';

export const mockGetUserInfoResponse: User = {
  userId: 1,
  email: 'user@example.com',
  nickname: 'Axel',
  icon: 'https://disneyparkstory.com/wp-content/uploads/2017/09/apple-touch-icon.png',
};
