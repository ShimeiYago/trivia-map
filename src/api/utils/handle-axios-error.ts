import { AxiosError } from 'axios';

export function handleAxiosError(axiosError: AxiosError): ApiError {
  if (axiosError.response) {
    // when API is active but response includes error
    const error: ApiError = {
      status: axiosError.response.status,
      data: axiosError.response.data,
      errorMsg: axiosError.message,
    };
    return error;
  } else {
    // when API is not active
    const error: ApiError = {
      status: 500,
      data: {},
      errorMsg: axiosError.message,
    };
    return error;
  }
}

export type ApiError = {
  status: number;
  data: unknown;
  errorMsg: string;
};
