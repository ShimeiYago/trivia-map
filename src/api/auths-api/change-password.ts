import { BASE_URL } from 'constant';
import { AxiosError } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';

export async function changePassword(
  password1: string,
  password2: string,
): Promise<void> {
  const axiosInstance = getAxiosInstance({}, {});

  const requestData: ChangePasswordRequest = {
    password1: password1,
    password2: password2,
  };

  try {
    // TODO: Set reasonable timeout
    await axiosInstance.post(`${BASE_URL}/auths/password/change`, requestData);
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError<ValidationError>(axiosError);
  }
}

export type ChangePasswordRequest = {
  password1: string;
  password2: string;
};

export type ValidationError = {
  password1?: string[];
  password2?: string[];
};
