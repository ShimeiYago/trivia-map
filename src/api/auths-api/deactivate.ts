import { API_TIMEOUT, BASE_URL } from 'constant';
import { AxiosError } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';

export async function deactivate(): Promise<void> {
  const axiosInstance = getAxiosInstance({ timeout: API_TIMEOUT.long }, {});

  try {
    await axiosInstance.put(`${BASE_URL}/auths/deactivate/`);
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}
