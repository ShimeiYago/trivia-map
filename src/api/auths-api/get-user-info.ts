import { BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { User } from 'types/user';
import { mockGetUserInfoResponse } from 'api/mock/auths-response/get-user-info';

export async function getUserInfo(): Promise<User> {
  const axiosInstance = getAxiosInstance({}, mockGetUserInfoResponse);

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<User> = await axiosInstance.get(`${BASE_URL}/auths/user`);
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}
