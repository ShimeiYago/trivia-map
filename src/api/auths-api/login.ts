import { BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockLoginResponse } from 'api/mock/auths-response/login';
import { User } from 'types/user';

export async function login(
  email: string,
  password: string,
): Promise<LoginResponse> {
  const axiosInstance = getAxiosInstance({}, mockLoginResponse);

  const requestData: LoginRequest = {
    email: email,
    password: password,
  };

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<LoginResponse> = await axiosInstance.post(
      `${BASE_URL}/auths/login`,
      requestData,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError<ValidationError>(axiosError);
  }
}

export type LoginRequest = {
  email: string;
  password: string;
};

export type LoginResponse = {
  access_token: string;
  refresh_token: string;
  user: User;
};

export type ValidationError = {
  email?: string[];
  password?: string[];
  non_field_errors?: string[];
};
