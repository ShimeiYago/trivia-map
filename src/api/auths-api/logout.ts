import { BASE_URL } from 'constant';
import { AxiosError } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';

export async function logout(): Promise<void> {
  const axiosInstance = getAxiosInstance({}, {});

  try {
    await axiosInstance.post(`${BASE_URL}/auths/logout`);
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}
