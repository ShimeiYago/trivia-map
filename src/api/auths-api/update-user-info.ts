import { API_TIMEOUT, BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { User } from 'types/user';
import { mockGetUserInfoResponse } from 'api/mock/auths-response/get-user-info';
import { SelializedImageFile } from 'types/selialized-image-file';
import { convertToFile } from 'utils/convert-to-file';

export async function updateUserInfo(param: {
  nickname: string;
  icon?: SelializedImageFile | null;
}): Promise<User> {
  const axiosInstance = getAxiosInstance({ timeout: API_TIMEOUT.long }, mockGetUserInfoResponse);

  const requestData = new FormData();
  requestData.append('nickname', param.nickname);

  switch (param.icon) {
    case undefined:
      break;
    case null:
      requestData.append('icon', '');
      break;
    default:
      const uploadFile = await convertToFile(param.icon);
      requestData.append('icon', uploadFile);
  }

  try {
    const res: AxiosResponse<User> = await axiosInstance.put(
      `${BASE_URL}/auths/user/update/`,
      requestData,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError<ValidationError>(axiosError);
  }
}

export type ValidationError = {
  nickname?: string[];
  icon?: string[];
};
