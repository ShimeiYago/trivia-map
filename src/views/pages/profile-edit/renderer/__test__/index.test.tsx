import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, Props, State } from '..';
import * as UpdateUserInfoModule from 'api/auths-api/update-user-info';
import { mockGetUserInfoResponse } from 'api/mock/auths-response/get-user-info';
import { ApiError } from 'api/utils/handle-axios-error';

let wrapper: ShallowWrapper<Props, State, Renderer>;

let updateUserInfoSpy: jest.SpyInstance;

const basicProps: Props = {
  user: {
    userId: 1,
    email: 'xxx@example.com',
    nickname: 'Axel',
    icon: 'https://...',
    isSocialAccount: false,
  },
  updateUser: jest.fn(),
  throwError: jest.fn(),
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<Renderer {...basicProps} />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('loading', () => {
    wrapper.setState({
      loadingState: 'loading',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('error case', () => {
    wrapper.setState({
      errorTitle: 'error title',
      errorMessages: ['error1', 'error2'],
      formError: {
        nickname: ['nickname is invalid'],
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('success', () => {
    wrapper.setState({
      loadingState: 'success',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('with icon as file', () => {
    wrapper.setState({
      icon: {
        dataUrl: 'data-url',
        fileName: 'file-name',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('with icon as null', () => {
    wrapper.setState({
      icon: null,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('social account', () => {
    wrapper.setProps({
      user: {
        userId: 1,
        email: '',
        nickname: 'Axel',
        icon: null,
        isSocialAccount: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('componentDidUpdate', () => {
  it('should set new user info', async () => {
    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    const prevProps = {
      user: undefined,
    } as Props;

    instance['componentDidUpdate'](prevProps);

    expect(instance.state.nickname).toBe('Axel');
  });

  it('should set no user info', async () => {
    const props = {
      user: undefined,
    } as Props;

    wrapper = shallow(<Renderer {...props} />);
    const instance = wrapper.instance();

    instance['componentDidUpdate'](basicProps);

    expect(instance.state.nickname).toBe('');
  });
});

describe('handleChangenickname', () => {
  it('should set nickname', async () => {
    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    const event = {
      target: {
        value: 'xxxxx',
      },
    } as React.ChangeEvent<HTMLInputElement>;

    instance['handleChangeNickname'](event);

    expect(instance.state.nickname).toBe('xxxxx');
  });
});

describe('handleSubmit', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    updateUserInfoSpy = jest.spyOn(UpdateUserInfoModule, 'updateUserInfo');
  });

  it('should set loadingState success when api succeed', async () => {
    updateUserInfoSpy.mockResolvedValue(mockGetUserInfoResponse);

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    await instance['handleSubmit']();

    expect(instance.state.loadingState).toBe('success');
  });

  it('should set loadingState success', async () => {
    updateUserInfoSpy.mockResolvedValue(mockGetUserInfoResponse);

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    await instance['handleSubmit']();

    expect(instance.state.loadingState).toBe('success');
  });

  it('should handle null image', async () => {
    updateUserInfoSpy.mockResolvedValue(mockGetUserInfoResponse);

    wrapper = shallow(<Renderer {...basicProps} />);
    wrapper.setState({
      icon: null,
      nickname: 'xxx',
    });
    const instance = wrapper.instance();

    await instance['handleSubmit']();

    expect(updateUserInfoSpy).toBeCalledWith({ icon: null, nickname: 'xxx' });
  });

  it('should set form error when api have validation error', async () => {
    const apiError: ApiError<UpdateUserInfoModule.ValidationError> = {
      status: 400,
      data: {
        nickname: ['nickname is invalid'],
      },
      errorMsg: '400 request is invalid',
    };
    updateUserInfoSpy.mockRejectedValue(apiError);

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    await instance['handleSubmit']();

    expect(instance.state.formError).toEqual({
      nickname: ['nickname is invalid'],
    });
  });

  it('should set loadingstate error when api fail', async () => {
    updateUserInfoSpy.mockRejectedValue(new Error());

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    await instance['handleSubmit']();

    expect(instance.state.loadingState).toBe('error');
  });
});

describe('handleChangeImage', () => {
  beforeEach(() => {
    wrapper = shallow(<Renderer {...basicProps} />);
  });

  it('set imageData when image is uploaded', async () => {
    const instance = wrapper.instance();

    const src = {
      dataUrl: 'data:image/png;base64,xxx',
      fileName: 'filename',
    };
    instance['handleChangeImage'](src);

    expect(instance.state.icon).toEqual({
      dataUrl: 'data:image/png;base64,xxx',
      fileName: 'filename',
    });
  });
});

describe('handleError', () => {
  beforeEach(() => {
    wrapper = shallow(<Renderer {...basicProps} />);
  });

  it('should call throwError prop', () => {
    const instance = wrapper.instance();
    instance['handleError']();

    expect(instance.props.throwError).toBeCalled();
  });
});
