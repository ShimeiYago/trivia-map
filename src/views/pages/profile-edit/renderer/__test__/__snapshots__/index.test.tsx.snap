// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<AdminWrapper>
  <ForwardRef(Stack)
    spacing={3}
  >
    <BackToNavi
      link="/admin/account"
      text="アカウント設定"
    />
    <ForwardRef(Box)>
      <ForwardRef(Stack)
        spacing={1}
        sx={
          Object {
            "maxWidth": "400px",
            "mx": "auto",
            "px": 1,
          }
        }
      >
        <ForwardRef(Typography)
          align="center"
          component="h2"
          variant="h5"
        >
          プロフィール変更
        </ForwardRef(Typography)>
        <ForwardRef(Box)>
          <ImageField
            cropable={true}
            disabled={false}
            error={false}
            maxLength={200}
            onCatchError={[Function]}
            onChange={[Function]}
            src="https://..."
            variant="icon"
          />
        </ForwardRef(Box)>
        <ForwardRef(Box)
          sx={
            Object {
              "pl": 1.5,
            }
          }
        >
          <ForwardRef(Typography)
            color="gray"
            fontSize={12}
            sx={
              Object {
                "pb": 0.5,
              }
            }
          >
            メールアドレス（非公開）
          </ForwardRef(Typography)>
          <ForwardRef(Typography)
            fontSize={18}
          >
            xxx@example.com
          </ForwardRef(Typography)>
        </ForwardRef(Box)>
        <ForwardRef(Box)
          component="form"
          noValidate={true}
        >
          <ForwardRef(TextField)
            disabled={false}
            error={false}
            fullWidth={true}
            id="nickname"
            inputProps={
              Object {
                "maxLength": 20,
              }
            }
            label="名前"
            margin="normal"
            name="nickname"
            onChange={[Function]}
            required={true}
            type="nickname"
            value="Axel"
          />
          <ForwardRef(TextField)
            disabled={false}
            error={false}
            fullWidth={true}
            id="url"
            label="Webサイト"
            margin="normal"
            name="url"
            onChange={[Function]}
            type="url"
            value="https://..."
          />
          <ForwardRef(LoadingButton)
            disabled={false}
            fullWidth={true}
            loading={false}
            onClick={[Function]}
            sx={
              Object {
                "mb": 2,
                "mt": 3,
              }
            }
            variant="contained"
          >
            更新
          </ForwardRef(LoadingButton)>
        </ForwardRef(Box)>
        <ForwardRef(Typography)
          align="right"
        >
          <Link
            to="/users/1"
          >
            プロフィール
          </Link>
        </ForwardRef(Typography)>
      </ForwardRef(Stack)>
    </ForwardRef(Box)>
  </ForwardRef(Stack)>
</AdminWrapper>
`;

exports[`Shallow Snapshot Tests error case 1`] = `
<AdminWrapper>
  <ForwardRef(Stack)
    spacing={3}
  >
    <BackToNavi
      link="/admin/account"
      text="アカウント設定"
    />
    <ForwardRef(Box)>
      <ForwardRef(Stack)
        spacing={1}
        sx={
          Object {
            "maxWidth": "400px",
            "mx": "auto",
            "px": 1,
          }
        }
      >
        <ForwardRef(Typography)
          align="center"
          component="h2"
          variant="h5"
        >
          プロフィール変更
        </ForwardRef(Typography)>
        <HeaderErrorMessages
          errorMessages={
            Array [
              "error1",
              "error2",
            ]
          }
          errorTitle="error title"
        />
        <ForwardRef(Box)>
          <ImageField
            cropable={true}
            disabled={false}
            error={false}
            maxLength={200}
            onCatchError={[Function]}
            onChange={[Function]}
            src="https://..."
            variant="icon"
          />
        </ForwardRef(Box)>
        <ForwardRef(Box)
          sx={
            Object {
              "pl": 1.5,
            }
          }
        >
          <ForwardRef(Typography)
            color="gray"
            fontSize={12}
            sx={
              Object {
                "pb": 0.5,
              }
            }
          >
            メールアドレス（非公開）
          </ForwardRef(Typography)>
          <ForwardRef(Typography)
            fontSize={18}
          >
            xxx@example.com
          </ForwardRef(Typography)>
        </ForwardRef(Box)>
        <ForwardRef(Box)
          component="form"
          noValidate={true}
        >
          <ForwardRef(TextField)
            disabled={false}
            error={true}
            fullWidth={true}
            helperText={
              Array [
                "nickname is invalid",
              ]
            }
            id="nickname"
            inputProps={
              Object {
                "maxLength": 20,
              }
            }
            label="名前"
            margin="normal"
            name="nickname"
            onChange={[Function]}
            required={true}
            type="nickname"
            value="Axel"
          />
          <ForwardRef(TextField)
            disabled={false}
            error={false}
            fullWidth={true}
            id="url"
            label="Webサイト"
            margin="normal"
            name="url"
            onChange={[Function]}
            type="url"
            value="https://..."
          />
          <ForwardRef(LoadingButton)
            disabled={false}
            fullWidth={true}
            loading={false}
            onClick={[Function]}
            sx={
              Object {
                "mb": 2,
                "mt": 3,
              }
            }
            variant="contained"
          >
            更新
          </ForwardRef(LoadingButton)>
        </ForwardRef(Box)>
        <ForwardRef(Typography)
          align="right"
        >
          <Link
            to="/users/1"
          >
            プロフィール
          </Link>
        </ForwardRef(Typography)>
      </ForwardRef(Stack)>
    </ForwardRef(Box)>
  </ForwardRef(Stack)>
</AdminWrapper>
`;

exports[`Shallow Snapshot Tests loading 1`] = `
<AdminWrapper>
  <ForwardRef(Stack)
    spacing={3}
  >
    <BackToNavi
      link="/admin/account"
      text="アカウント設定"
    />
    <ForwardRef(Box)>
      <ForwardRef(Stack)
        spacing={1}
        sx={
          Object {
            "maxWidth": "400px",
            "mx": "auto",
            "px": 1,
          }
        }
      >
        <ForwardRef(Typography)
          align="center"
          component="h2"
          variant="h5"
        >
          プロフィール変更
        </ForwardRef(Typography)>
        <ForwardRef(Box)>
          <ImageField
            cropable={true}
            disabled={true}
            error={false}
            maxLength={200}
            onCatchError={[Function]}
            onChange={[Function]}
            src="https://..."
            variant="icon"
          />
        </ForwardRef(Box)>
        <ForwardRef(Box)
          sx={
            Object {
              "pl": 1.5,
            }
          }
        >
          <ForwardRef(Typography)
            color="gray"
            fontSize={12}
            sx={
              Object {
                "pb": 0.5,
              }
            }
          >
            メールアドレス（非公開）
          </ForwardRef(Typography)>
          <ForwardRef(Typography)
            fontSize={18}
          >
            xxx@example.com
          </ForwardRef(Typography)>
        </ForwardRef(Box)>
        <ForwardRef(Box)
          component="form"
          noValidate={true}
        >
          <ForwardRef(TextField)
            disabled={true}
            error={false}
            fullWidth={true}
            id="nickname"
            inputProps={
              Object {
                "maxLength": 20,
              }
            }
            label="名前"
            margin="normal"
            name="nickname"
            onChange={[Function]}
            required={true}
            type="nickname"
            value="Axel"
          />
          <ForwardRef(TextField)
            disabled={true}
            error={false}
            fullWidth={true}
            id="url"
            label="Webサイト"
            margin="normal"
            name="url"
            onChange={[Function]}
            type="url"
            value="https://..."
          />
          <ForwardRef(LoadingButton)
            disabled={true}
            fullWidth={true}
            loading={true}
            onClick={[Function]}
            sx={
              Object {
                "mb": 2,
                "mt": 3,
              }
            }
            variant="contained"
          >
            更新
          </ForwardRef(LoadingButton)>
        </ForwardRef(Box)>
        <ForwardRef(Typography)
          align="right"
        >
          <Link
            to="/users/1"
          >
            プロフィール
          </Link>
        </ForwardRef(Typography)>
      </ForwardRef(Stack)>
    </ForwardRef(Box)>
  </ForwardRef(Stack)>
</AdminWrapper>
`;

exports[`Shallow Snapshot Tests social account 1`] = `
<AdminWrapper>
  <ForwardRef(Stack)
    spacing={3}
  >
    <BackToNavi
      link="/admin/account"
      text="アカウント設定"
    />
    <ForwardRef(Box)>
      <ForwardRef(Stack)
        spacing={1}
        sx={
          Object {
            "maxWidth": "400px",
            "mx": "auto",
            "px": 1,
          }
        }
      >
        <ForwardRef(Typography)
          align="center"
          component="h2"
          variant="h5"
        >
          プロフィール変更
        </ForwardRef(Typography)>
        <ForwardRef(Alert)
          severity="info"
        >
          ソーシャルアカウントを連携している場合、このページからプロフィール情報を変更することはできません。表示されている情報が古い場合は、一度ログアウト後再びログインしてください。
        </ForwardRef(Alert)>
        <ForwardRef(Box)>
          <ImageField
            cropable={true}
            disabled={true}
            error={false}
            maxLength={200}
            onCatchError={[Function]}
            onChange={[Function]}
            variant="icon"
          />
        </ForwardRef(Box)>
        <ForwardRef(Box)
          component="form"
          noValidate={true}
        >
          <ForwardRef(TextField)
            disabled={true}
            error={false}
            fullWidth={true}
            id="nickname"
            inputProps={
              Object {
                "maxLength": 20,
              }
            }
            label="名前"
            margin="normal"
            name="nickname"
            onChange={[Function]}
            required={true}
            type="nickname"
            value="Axel"
          />
          <ForwardRef(TextField)
            disabled={true}
            error={false}
            fullWidth={true}
            id="url"
            label="Webサイト"
            margin="normal"
            name="url"
            onChange={[Function]}
            type="url"
            value=""
          />
          <ForwardRef(LoadingButton)
            disabled={true}
            fullWidth={true}
            loading={false}
            onClick={[Function]}
            sx={
              Object {
                "mb": 2,
                "mt": 3,
              }
            }
            variant="contained"
          >
            更新
          </ForwardRef(LoadingButton)>
        </ForwardRef(Box)>
        <ForwardRef(Typography)
          align="right"
        >
          <Link
            to="/users/1"
          >
            プロフィール
          </Link>
        </ForwardRef(Typography)>
      </ForwardRef(Stack)>
    </ForwardRef(Box)>
  </ForwardRef(Stack)>
</AdminWrapper>
`;

exports[`Shallow Snapshot Tests success 1`] = `
<AdminWrapper>
  <ForwardRef(Stack)
    spacing={3}
  >
    <BackToNavi
      link="/admin/account"
      text="アカウント設定"
    />
    <ForwardRef(Box)>
      <ForwardRef(Stack)
        spacing={1}
        sx={
          Object {
            "maxWidth": "400px",
            "mx": "auto",
            "px": 1,
          }
        }
      >
        <ForwardRef(Typography)
          align="center"
          component="h2"
          variant="h5"
        >
          プロフィール変更
        </ForwardRef(Typography)>
        <ForwardRef(Alert)>
          ユーザー情報を変更しました。
        </ForwardRef(Alert)>
        <ForwardRef(Box)>
          <ImageField
            cropable={true}
            disabled={false}
            error={false}
            maxLength={200}
            onCatchError={[Function]}
            onChange={[Function]}
            src="https://..."
            variant="icon"
          />
        </ForwardRef(Box)>
        <ForwardRef(Box)
          sx={
            Object {
              "pl": 1.5,
            }
          }
        >
          <ForwardRef(Typography)
            color="gray"
            fontSize={12}
            sx={
              Object {
                "pb": 0.5,
              }
            }
          >
            メールアドレス（非公開）
          </ForwardRef(Typography)>
          <ForwardRef(Typography)
            fontSize={18}
          >
            xxx@example.com
          </ForwardRef(Typography)>
        </ForwardRef(Box)>
        <ForwardRef(Box)
          component="form"
          noValidate={true}
        >
          <ForwardRef(TextField)
            disabled={false}
            error={false}
            fullWidth={true}
            id="nickname"
            inputProps={
              Object {
                "maxLength": 20,
              }
            }
            label="名前"
            margin="normal"
            name="nickname"
            onChange={[Function]}
            required={true}
            type="nickname"
            value="Axel"
          />
          <ForwardRef(TextField)
            disabled={false}
            error={false}
            fullWidth={true}
            id="url"
            label="Webサイト"
            margin="normal"
            name="url"
            onChange={[Function]}
            type="url"
            value="https://..."
          />
          <ForwardRef(LoadingButton)
            disabled={false}
            fullWidth={true}
            loading={false}
            onClick={[Function]}
            sx={
              Object {
                "mb": 2,
                "mt": 3,
              }
            }
            variant="contained"
          >
            更新
          </ForwardRef(LoadingButton)>
        </ForwardRef(Box)>
        <ForwardRef(Typography)
          align="right"
        >
          <Link
            to="/users/1"
          >
            プロフィール
          </Link>
        </ForwardRef(Typography)>
      </ForwardRef(Stack)>
    </ForwardRef(Box)>
  </ForwardRef(Stack)>
</AdminWrapper>
`;

exports[`Shallow Snapshot Tests with icon as file 1`] = `
<AdminWrapper>
  <ForwardRef(Stack)
    spacing={3}
  >
    <BackToNavi
      link="/admin/account"
      text="アカウント設定"
    />
    <ForwardRef(Box)>
      <ForwardRef(Stack)
        spacing={1}
        sx={
          Object {
            "maxWidth": "400px",
            "mx": "auto",
            "px": 1,
          }
        }
      >
        <ForwardRef(Typography)
          align="center"
          component="h2"
          variant="h5"
        >
          プロフィール変更
        </ForwardRef(Typography)>
        <ForwardRef(Box)>
          <ImageField
            cropable={true}
            disabled={false}
            error={false}
            maxLength={200}
            onCatchError={[Function]}
            onChange={[Function]}
            src="data-url"
            variant="icon"
          />
        </ForwardRef(Box)>
        <ForwardRef(Box)
          sx={
            Object {
              "pl": 1.5,
            }
          }
        >
          <ForwardRef(Typography)
            color="gray"
            fontSize={12}
            sx={
              Object {
                "pb": 0.5,
              }
            }
          >
            メールアドレス（非公開）
          </ForwardRef(Typography)>
          <ForwardRef(Typography)
            fontSize={18}
          >
            xxx@example.com
          </ForwardRef(Typography)>
        </ForwardRef(Box)>
        <ForwardRef(Box)
          component="form"
          noValidate={true}
        >
          <ForwardRef(TextField)
            disabled={false}
            error={false}
            fullWidth={true}
            id="nickname"
            inputProps={
              Object {
                "maxLength": 20,
              }
            }
            label="名前"
            margin="normal"
            name="nickname"
            onChange={[Function]}
            required={true}
            type="nickname"
            value="Axel"
          />
          <ForwardRef(TextField)
            disabled={false}
            error={false}
            fullWidth={true}
            id="url"
            label="Webサイト"
            margin="normal"
            name="url"
            onChange={[Function]}
            type="url"
            value="https://..."
          />
          <ForwardRef(LoadingButton)
            disabled={false}
            fullWidth={true}
            loading={false}
            onClick={[Function]}
            sx={
              Object {
                "mb": 2,
                "mt": 3,
              }
            }
            variant="contained"
          >
            更新
          </ForwardRef(LoadingButton)>
        </ForwardRef(Box)>
        <ForwardRef(Typography)
          align="right"
        >
          <Link
            to="/users/1"
          >
            プロフィール
          </Link>
        </ForwardRef(Typography)>
      </ForwardRef(Stack)>
    </ForwardRef(Box)>
  </ForwardRef(Stack)>
</AdminWrapper>
`;

exports[`Shallow Snapshot Tests with icon as null 1`] = `
<AdminWrapper>
  <ForwardRef(Stack)
    spacing={3}
  >
    <BackToNavi
      link="/admin/account"
      text="アカウント設定"
    />
    <ForwardRef(Box)>
      <ForwardRef(Stack)
        spacing={1}
        sx={
          Object {
            "maxWidth": "400px",
            "mx": "auto",
            "px": 1,
          }
        }
      >
        <ForwardRef(Typography)
          align="center"
          component="h2"
          variant="h5"
        >
          プロフィール変更
        </ForwardRef(Typography)>
        <ForwardRef(Box)>
          <ImageField
            cropable={true}
            disabled={false}
            error={false}
            maxLength={200}
            onCatchError={[Function]}
            onChange={[Function]}
            variant="icon"
          />
        </ForwardRef(Box)>
        <ForwardRef(Box)
          sx={
            Object {
              "pl": 1.5,
            }
          }
        >
          <ForwardRef(Typography)
            color="gray"
            fontSize={12}
            sx={
              Object {
                "pb": 0.5,
              }
            }
          >
            メールアドレス（非公開）
          </ForwardRef(Typography)>
          <ForwardRef(Typography)
            fontSize={18}
          >
            xxx@example.com
          </ForwardRef(Typography)>
        </ForwardRef(Box)>
        <ForwardRef(Box)
          component="form"
          noValidate={true}
        >
          <ForwardRef(TextField)
            disabled={false}
            error={false}
            fullWidth={true}
            id="nickname"
            inputProps={
              Object {
                "maxLength": 20,
              }
            }
            label="名前"
            margin="normal"
            name="nickname"
            onChange={[Function]}
            required={true}
            type="nickname"
            value="Axel"
          />
          <ForwardRef(TextField)
            disabled={false}
            error={false}
            fullWidth={true}
            id="url"
            label="Webサイト"
            margin="normal"
            name="url"
            onChange={[Function]}
            type="url"
            value="https://..."
          />
          <ForwardRef(LoadingButton)
            disabled={false}
            fullWidth={true}
            loading={false}
            onClick={[Function]}
            sx={
              Object {
                "mb": 2,
                "mt": 3,
              }
            }
            variant="contained"
          >
            更新
          </ForwardRef(LoadingButton)>
        </ForwardRef(Box)>
        <ForwardRef(Typography)
          align="right"
        >
          <Link
            to="/users/1"
          >
            プロフィール
          </Link>
        </ForwardRef(Typography)>
      </ForwardRef(Stack)>
    </ForwardRef(Box)>
  </ForwardRef(Stack)>
</AdminWrapper>
`;
